### ISSUE TITLE
Add preserve inline function for expand style braces

### ISSUE THREAD

**Issue by Kutsan** [beautify-web/js-beautify#1052]
# Description
Basically, I want `collapse-preserve-inline` feature for expand braces style like `expand-preserve-inline` which is doesn't exist at the moment.

# Input
The code looked like this before beautification:
```javascript
foo() {
    var a = 2;
}
```

# Expected Output
The code should have looked like this after beautification:
```javascript
foo() { var a = 2; }
```

# Actual Output
The  code actually looked like this after beautification:
```javascript
foo()
{
    var a = 2;
}
```

**Comment by bitwiseman** [comment]
@Kutsan -  `expand-preserve inline` would look like this:
The code looked like this before beautification:
```javascript
function foo() { var a = 2; }
function bar() { 
    var a = 2; 
}
```

The  code would look like this after beautification:
```javascript
function foo() { var a = 2; }
function bar() 
{ 
    var a = 2; 
}
```

I'm not sure what you're describing - I need more examples.  Do you want all braces collapsed to inline if possible?  Or do you want expanded braces to be kept but non-expanded to be inlined?

**Comment by Coburn37** [comment]
Just leaving a comment as you replied, haha. Yeah, this is exactly what I'm looking for.

I think this is just an expansion of the solution implemented in #382 as to fix import and export newlines you need to use `collapse-preserve-inline` which limits you to just `collapse` brace styling. It would be nice to have a `preserve-inline-braces` option which can be used with any of the braces styling formats.

**Comment by Coburn37** [comment]
Okay, I made #1058 . Now that I really look at it, I don't think this fixes your problem. It only preserves functions that already look like your expected output, it does not roll up function from your input to your expected output so I am of no help to you :(
